src/chain/chain_information/babe_fetch_epoch.rs

```diff
@@ -39,6 +39,8 @@ pub struct Config {
     pub runtime: host::HostVmPrototype,
     /// The Babe epoch to fetch.
     pub epoch_to_fetch: BabeEpochToFetch,
+    /// The Babe genesis config.
+    pub babe_genesis_config: header::BabeNextConfig,
```

Comment:
The objective of the `babe_fetch_epoch` module is to do a runtime call.
You're passing a `BabeNextConfig` here just so that you can later return back what it contains.

Please define a new `EpochInfo` struct in this module and return it, instead of doing that.

```suggestion
```